spec/ref_arch_setup/install_spec.rb

```diff
@@ -137,25 +204,561 @@
+          expect { install.handle_tarball_path(pe_tarball) }
+            .to raise_error(RuntimeError)
+        end
+      end
+    end
+
+    context "when the target master is not localhost" do
+      before do
+        install.instance_variable_set(:@target_master, remote_target_master)
+      end
+
+      context "when the tarball is handled successfully" do
+        it "returns the filename" do
+          expect(File).to receive(:basename).with(pe_tarball).and_return(pe_tarball_filename)
+          expect(install).to receive(:target_master_is_localhost?).and_return(false)
+
+          expect(install).to receive(:handle_tarball_path_with_remote_target_master)
+            .with(pe_tarball).and_return(true)
+
+          expect(install.handle_tarball_path(pe_tarball))
+            .to eq(tarball_path_on_master)
+        end
+      end
+
+      context "when the tarball is not handled successfully" do
+        it "raises an error" do
+          expect(File).to receive(:basename).with(pe_tarball).and_return(pe_tarball_filename)
+          expect(install).to receive(:target_master_is_localhost?).and_return(false)
+
+          expect(install).to receive(:handle_tarball_path_with_remote_target_master)
+            .with(pe_tarball).and_return(false)
+
+          expect { install.handle_tarball_path(pe_tarball) }
+            .to raise_error(RuntimeError)
+        end
+      end
+    end
+  end
+
+  describe "#handle_pe_tarball" do
+    context "when the extension is valid" do
+      context "when the tarball path is a valid URL" do
+        context "when the tarball URL is handled successfully" do
+          it "returns the tarball path on the master" do
+            # expect(install).to receive(:validate_extension).with(pe_tarball_url)
+            expect(install).to receive(:valid_url?).with(pe_tarball_url).and_return(true)
+            expect(install).to receive(:handle_tarball_url)
+              .with(pe_tarball_url).and_return(tarball_path_on_master)
+
+            expect(install.handle_pe_tarball(pe_tarball_url))
+              .to eq(tarball_path_on_master)
+          end
+        end
+
+        context "when the tarball URL is not handled successfully" do
+          it "raises an error" do
+            # expect(install).to receive(:validate_extension).with(pe_tarball_url)
+            expect(install).to receive(:valid_url?).with(pe_tarball_url).and_return(true)
+            expect(install).to receive(:handle_tarball_url).with(pe_tarball_url).and_return(nil)
+
+            expect { install.handle_pe_tarball(pe_tarball_url) }
+              .to raise_error(RuntimeError)
+          end
+        end
+      end
+
+      context "when the tarball path is not a valid URL and a path is assumed" do
+        context "when the tarball path is handled successfully" do
+          it "returns the tarball path on the master" do
+            # expect(install).to receive(:validate_extension).with(pe_tarball)
```

Comment:
I think the comment can be removed.

Reply:
Updated to uncomment these and make them work.